crates/wasi/src/preview2/poll.rs

```diff
@@ -116,19 +116,24 @@ impl<T: WasiView> poll::Host for T {
 
         Ok(PollList { futures }.await)
     }
+}
 
-    async fn poll_one(&mut self, pollable: Resource<Pollable>) -> Result<()> {
+#[async_trait::async_trait]
+impl<T: WasiView> crate::preview2::bindings::io::poll::HostPollable for T {
+    async fn block(&mut self, pollable: Resource<Pollable>) -> Result<()> {
         let table = self.table_mut();
-
         let pollable = table.get(&pollable)?;
         let ready = (pollable.make_future)(table.get_any_mut(pollable.index)?);
         ready.await;
         Ok(())
     }
-}
-
-#[async_trait::async_trait]
-impl<T: WasiView> crate::preview2::bindings::io::poll::HostPollable for T {
+    async fn ready(&mut self, pollable: Resource<Pollable>) -> Result<bool> {
+        let table = self.table_mut();
+        let pollable = table.get(&pollable)?;
+        let ready = (pollable.make_future)(table.get_any_mut(pollable.index)?);
+        futures::pin_mut!(ready);
+        Ok(futures::future::poll_immediate(ready).await.is_some())
```

Comment:
Could this perhaps be `matches!(..., Some(()))` to emphasize this isn't accidentally throwing away a "real" result from the poll?

Reply:
Good idea!